core/domain/improvements_domain.py

```diff
@@ -53,8 +53,40 @@ class TaskEntry(python_utils.OBJECT):
 
     def __init__(
             self, entity_type, entity_id, entity_version, task_type,
-            target_type, target_id, issue_description, status, resolver_id,
-            resolved_on):
+            target_type, target_id, issue_description, status, resolver_id=None,
+            resolved_on=None):
```

Comment:
Why not always pass in the resolver_id and resolved_on? In general avoid default args where possible, to reduce branching (it helps maintainability).

Reply:
Done.